src/database.rs

```diff
@@ -145,21 +145,50 @@ impl Database {
         group_id: usize,
         commitment: &Hash,
         block_number: usize,
-        tree_idx: usize,
+        relative_index: i64,
     ) -> Result<(), Error> {
         let query = sqlx::query(
             r#"UPDATE pending_identities
-                   SET mined_in_block = $1, assigned_leaf_idx = $2
+                   SET mined_in_block = $1, insertion_idx = $2
                    WHERE group_id = $3 AND commitment = $4;"#,
         )
         .bind(block_number as i64)
-        .bind(tree_idx as i64)
+        .bind(relative_index)
         .bind(group_id as i64)
         .bind(commitment);
+
         self.pool.execute(query).await?;
         Ok(())
     }
 
+    pub async fn pending_identity_confirmed(
```

Comment:
2 things wrong with this IMO:

1. It lacks an active verb in the name – something like `mark_pending_identity_confirmed` or something to that effect – this name doesn't tell me "I'm about to mutate something".
2. It hides a lot of complexity behind this name – the current name only allows you to remove a pending item, not to invalidate all previous names. Change the name of this, or split it into two methods, so that I'm not surprised when an innocuously named method queues a bunch of work.

Reply:
Renamed to `confirm_identity_and_retrigger_stale_recods`